src/analysis/buildTypeWeights.ts

```diff
@@ -30,6 +63,145 @@ function buildTypeWeightsFromSchema(
+                    weight: typeWeights.object || DEFAULT_OBJECT_WEIGHT,
+                };
+            }
+        }
+    });
+
+    // Get any Query fields (these are the queries that the API exposes)
+    const queryType: Maybe<GraphQLObjectType> = schema.getQueryType();
+
+    if (queryType) {
+        result.Query = {
+            weight: typeWeights.query || DEFAULT_QUERY_WEIGHT,
+            // fields gets populated with the query fields and associated weights.
+            fields: {},
+        };
+
+        const queryFields: GraphQLFieldMap<any, any> = queryType.getFields();
+
+        Object.keys(queryFields).forEach((field) => {
+            // this is the type the query resolves to
+            const resolveType: GraphQLOutputType = queryFields[field].type;
+
+            // check if any of our keywords 'first', 'last', 'limit' exist in the arg list
+            queryFields[field].args.forEach((arg: GraphQLArgument) => {
+                // If query has an argument matching one of the limiting keywords and resolves to a list then the weight of the query
+                // should be dependent on both the weight of the resolved type and the limiting argument.
+                if (KEYWORDS.includes(arg.name) && isListType(resolveType)) {
+                    const defaultVal: number = <number>arg.defaultValue;
+
+                    // Get the type that comprises the list
+                    const listType = resolveType.ofType;
+
+                    // Composite Types are Objects, Interfaces and Unions.
+                    if (isCompositeType(listType)) {
+                        // Set the field weight to a function that accepts
+                        // TODO: Accept ArgumentNode[] and look for the arg we need.
+                        // TODO: Test this function
+                        result.Query.fields[field] = (args: ArgumentNode[]): number => {
+                            // Function should receive object with arg, value as k, v pairs
+                            // function iterate on this object looking for a keyword then returns
+                            const limitArg: ArgumentNode | undefined = args.find(
+                                (cur) => cur.name.value === arg.name
+                            );
+
+                            // const isVariable = (node: any): node is VariableNode => {
+                            //     if (node as VariableNode) return true;
+                            //     return false;
+                            // };
+
+                            const isIntNode = (node: any): node is IntValueNode => {
+                                if (node as IntValueNode) return true;
+                                return false;
+                            };
+
+                            if (limitArg) {
+                                const node: ValueNode = limitArg.value;
+
+                                // FIXME: Is there a better way to check for the type here?
```

Comment:
You can use the `Kind` enum from GraphQL. try `Kind.INT`  to check for `"IntValue"`. It's worknig pretty well in the complexity algo.

Reply:
Working with schemas may be different than working with query ASTs though...

Reply:
Didn't know about these. Thank you!